src/gui/main_screen.py

```diff
@@ -0,0 +1,483 @@
+                          int(window_height / self.zoom_level - (self.pan_y / self.zoom_level)),
+                          int(-self.pan_x / self.zoom_level):
+                          int(window_width / self.zoom_level - (self.pan_x / self.zoom_level))
+                          ]
+            # TODO slicing image array has to be proportional to the original proportions of the image to retain readability
+            #  and allow for rectangle images
+
+            self.img = Image.fromarray(image_array)
+            self.img = self.img.resize((min(window_width, self.zoomed_width),
+                                        min(window_height, self.zoomed_height)),
+                                       Image.NEAREST)
+
+            self.current_image = ImageTk.PhotoImage(self.img)
+            self.image_id = self.create_image(max(0, (window_width - self.zoomed_width) // 2),
+                                              max(0, (window_height - self.zoomed_height) // 2),
+                                              anchor=tk.NW,
+                                              image=self.current_image)
+
+        def on_mousewheel(self, event):
+            zoom_factor = 1.1 if event.delta > 0 else 10 / 11
+
+            window_width = self.winfo_width()
+            window_height = self.winfo_height()
+
+            before_x = (self.image_array_width - (window_width / self.zoom_level))
+            before_y = (self.image_array_height - (window_height / self.zoom_level))
+
+            self.zoom_level *= zoom_factor
+            self.zoom_level = min(max(self.zoom_level, self.initial_zoom_level), 100)
+
+            if event.delta > 0:
+                self.pan_x -= max(
+                    (((self.image_array_width - (window_width / self.zoom_level)) - before_x) / 2) * self.zoom_level, 0)
+                self.pan_y -= max(
+                    (((self.image_array_height - (window_height / self.zoom_level)) - before_y) / 2) * self.zoom_level,
+                    0)
+            else:
+                self.pan_x += max(
+                    ((before_x - (self.image_array_width - (window_width / self.zoom_level))) / 2) * self.zoom_level, 0)
+                self.pan_y += max(
+                    ((before_y - (self.image_array_height - (window_height / self.zoom_level))) / 2) * self.zoom_level,
+                    0)
+
+            shift_x = event.x / window_width - 0.5
+            shift_y = event.y / window_height - 0.5
+
+            self.pan_x -= shift_x * window_width
+            self.pan_y -= shift_y * window_height
+
+            self.update_image()
+            self.image_change_controller.update_zoom_infobox_value()
+
+        def on_button_press(self, event):
+            self.prev_x = event.x
+            self.prev_y = event.y
+            self.is_holding = True
+
+        def on_button_release(self, event):
+            self.is_holding = False
+            if self.is_panning:
```

Comment:
Early returns could help there.
I guess we should try to avoid code with a lot of indentations